[IMP] POS: payment in self-ordering mode (#19417)
odoo/documentation: content/applications/sales/point_of_sale/self_order.rst

Reviewer: _#iknowyoudidn'tupdatethisbut_

Since this really does not look nice as a tip, and as it's really not so complicated that it needs to be broken into steps, would you mind doing this:

```suggestion
.. tip::
To customize QR codes, scan the relevant QR code to acquire its URL, then use a QR code generator (e.g., `QR code monkey <https://www.qrcode-monkey.com>`_ or `QR code generator <https://www.qr-code-generator.com>`_) to create a custom QR code.
```

Self-ordering
=============

The self-ordering mode allows customers to browse the menu or product catalog, place an order,
and complete payment using their mobile device or a self-ordering kiosk.

.. _pos/self_order/configuration:

Configuration
=============

To enable the self-ordering mode, access the :ref:`POS settings <configuration/settings>`, scroll
down to the :guilabel:`Mobile self-order & Kiosk` section, and, in
the :guilabel:`QR menu & Kiosk activation` section, select a :guilabel:`Self Ordering` type from the
dropdown list.

You can choose between :guilabel:`QR menu`, :guilabel:`QR menu + Ordering`, and :guilabel:`Kiosk`.

.. tabs::

>    .. group-tab:: QR menu

Reviewer: Shouldn't this be:

```suggestion
   .. group-tab:: QR menu or QR menu + Ordering
```

      Select :guilabel:`QR menu` or :guilabel:`QR menu + Ordering` to give customers access to the
      menu or product catalog by scanning a QR code on their personal device. The latter also
      allows them to place an order and make a payment. Then:

      - Click :icon:`fa-arrow-right` :guilabel:`Print QR Codes` to download a .pdf document with the
        generated QR codes.
      - Click :icon:`fa-arrow-right` :guilabel:`Download QR Codes` to download a compressed file
        with the generated QR codes.

>       - Click :icon:`fa-arrow-right` :guilabel:`Free Metal / Wood Stands` to `order QR codes stands

Reviewer: ```suggestion
      - Click :icon:`fa-arrow-right` :guilabel:`Free Metal / Wood Stands` to `order QR code stands
```

   .. group-tab:: Kiosk

      Select :guilabel:`Kiosk` to let customers browse the menu or product catalog, place orders,
      and pay directly at a self-ordering kiosk.

Once a self-ordering type is selected, click :icon:`fa-arrow-right` :guilabel:`Preview Web
interface` under the :guilabel:`Self  Ordering` field to ensure all :ref:`additional settings
<pos/self_order/additional-settings>` are correctly applied.

.. _pos/self_order/additional-settings:

Additional settings
-------------------

To further configure the self-ordering options, use the settings in the :guilabel:`Mobile self-order

> & Kiosk`.

Reviewer: ```suggestion
To further configure the self-ordering options, use the settings in the :guilabel:`Mobile self-order
& Kiosk` section.
```

      .. note::
         - Leaving the :guilabel:`Points of Sale` field empty shares the button with all POS.
         - The :guilabel:`Preview` column automatically updates, displaying a glimpse of the

>            button's appearance based on its configuration.

Reviewer: For me, we should take this opportunity to remove 'glimpse' as (IMO) it's not suited to user documentation (and also, it's typically used for a more momentary or partial view of something). How about:

```suggestion
         - The :guilabel:`Preview` column automatically updates to show the button's appearance for the selected :guilabel:`Style`.
```

   .. tab:: Service location and payment options

      In the :guilabel:`Service at` field, specify where the service takes place by selecting
      :guilabel:`Table` or :guilabel:`Pickup zone`. Then, in the :guilabel:`Pay after` field, define
      when customers pay. Depending on the self-ordering type, service location, and POS type,

>       customers can pay after :guilabel:`Each meal` or :guilabel:`Each order` and choose from

Reviewer: ```suggestion
      customers can pay after their :guilabel:`Meal` or after :guilabel:`Each order` and choose from
```

Just going by the UI: 

<img width="438" height="159" alt="Image" src="https://github.com/user-attachments/assets/d267ceae-3a68-4fb2-950b-90544c0aac31" />

(And also doesn't really make sense to talk about customers paying after each meal, like they'd be having multiple meals.)

Author: You are totally right. I don't know why I used "Each meal"

      different payment options.

      .. tabs::

         .. group-tab:: QR menu

            - **Restaurants**: Customers can be served at their :guilabel:`Table` or the

>               :guilabel:`Pickup zone`.

Reviewer: ```suggestion
            - **Restaurants**: Customers can be served at their :guilabel:`Table` or at the
              :guilabel:`Pickup zone`.
```

>               - For :guilabel:`Table` service, they can pay after :guilabel:`Each meal` or

Reviewer: ```suggestion
              - For :guilabel:`Table` service, they can pay after their :guilabel:`Meal` or after
```

                :guilabel:`Each order`.
              - For :guilabel:`Pickup zone` service, they can only pay after :guilabel:`Each order`.

            - **Shops**: Customers can only be served at the :guilabel:`Pickup zone` and pay after
              :guilabel:`Each order`.
            - Customers can pay :doc:`online </applications/finance/payment_providers>` or using any
              configured :doc:`payment method <payment_methods>`. If the :guilabel:`Online Payment`
              field is left empty, customers pay at the cashier.

         .. group-tab:: Kiosk

            - Customers can either be served at their :guilabel:`Table` or the :guilabel:`Pickup

>               zone`, but they must pay after :guilabel:`Each order`.

Reviewer: ```suggestion
            - Customers can be served at their :guilabel:`Table` or at the :guilabel:`Pickup
              zone`, but they must pay after :guilabel:`Each order`.
```

            - The kiosk self-ordering only works with :doc:`Adyen <payment_methods/terminals/adyen>`

>               and :doc:`Stripe <payment_methods/terminals/stripe>` terminals.

Reviewer: Could this be interpreted as 'online payment is not possible'?
Maybe we could rephrase it as:

```suggestion
            - Payment via a terminal is only possible with :doc:`Adyen <payment_methods/terminals/adyen>` or :doc:`Stripe <payment_methods/terminals/stripe>` terminals.
```

Author: No, online payment is possible. This was actually the reason why I checked the page. You can select an online payment option and use a QR code to pay. But I like your suggestion for this sentence.

            - To pay :doc:`online </applications/finance/payment_providers>`, customers must scan
              the displayed :doc:`QR code

>               </applications/sales/point_of_sale/payment_methods/qr_code_payment>`.

Reviewer: Could you help me understand this? (Like, do we need to be directing the user to the 'applications/finance/payment_providers' page first? Or do they only have to follow instructions on the QR code payment page to set that up? Feeling like I never fully grasp all this payments stuff 😕 )

Author: You have to set up an online payment method to display the QR code for online payment. You can then scan the QR code to choose a provider. For example, I added Demo, PayPal, and Adyen as providers. Once I scan the QR code, I get redirected to a page where I can choose between them. But you are right. It is confusing to add the link to the QR code payment page, as this is a different workflow.

>             - :guilabel:`Cash` payment methods are not supported.

Reviewer: Could this just be:

```suggestion
            - :guilabel:`Cash` payments are not supported.
```

      .. note::
         Regardless of the self-ordering configuration, only one :guilabel:`Online Payment` method
         can be added.

      .. tip::
         Click :icon:`fa-arrow-right` :guilabel:`Payment Methods` to :ref:`create or edit payment
         methods <pos/qr_code_payment/create-method>`.

      .. seealso::
         - :doc:`../../finance/payment_providers`
         - :doc:`payment_methods`

   .. tab:: Language

>       Enable multiple languages for the self-ordering interface. The  suggested languages are those

Reviewer: ```suggestion
      Enable multiple languages for the self-ordering interface. The suggested languages are those
```

      already installed in Odoo. When several languages are installed, select a :guilabel:`Default`

>       language. To add more languages, follow these steps:

Reviewer: I guess if several languages are installed you can already select a default and available.

So maybe:

```suggestion
      already installed in Odoo. When several languages are installed, select a :guilabel:`Default`
      language and, if desired, additional :guilabel:`Available` languages. To add more languages, follow these steps:
```

      .. note::
         You can add multiple splash screen images at once.

   .. tab:: Header

      Add a custom image to the self-ordering header to give it a branded look. To do so, go to the
      :guilabel:`Customize Header` section and click :guilabel:`Upload your file`.

.. _pos/self_order/use:

Using the self-ordering mode
============================

Once the self-ordering configuration is completed, launch the self-ordering interface from the POS

> and make it available to customers.

Reviewer: I'm afraid I don't follow what the user needs to do here. Could you explain / show me?

Author: You are right! I will rephrase it.

         - Click the provided URL to open the kiosk in a new tab;
         - Click :guilabel:`Install App` to install the kiosk module on your self-ordering kiosk; or
         - Click :guilabel:`Open on IoT Box` if your kiosk is :doc:`connected to an IoT system
           <../../general/iot/connect>`.

      .. note::
         - Once a session is open, :guilabel:`Start Kiosk` switches to :guilabel:`Open Kiosk` on the
           POS card.
         - Click :guilabel:`Open Kiosk` on the POS card to reopen the popover and access the
           self-ordering interface.

      On the customers' end,

      #. Press the :ref:`home button <pos/self_order/additional-settings>` on the self-ordering
         kiosk to open the menu or product catalog.
      #. Select the items, then click :guilabel:`Order` to place an order.

>       #. Follow the instructions on-screen to assign a table and pay for the order.

Reviewer: Earlier it said that for kiosk, 'Customers can either be served at their Table or the Pickup zone' ... So maybe this needs to be something like:

```suggestion
      #. Follow the instructions on-screen to assign a table, if relevant, and pay for the order.
```

Author: Good point! :)
